src/NUnitFramework/framework/Constraints/EqualityAdapter.cs

```diff
@@ -193,6 +199,11 @@ class EqualityComparerAdapter<T> : GenericEqualityAdapter<T>
         {
             private readonly IEqualityComparer<T> comparer;
 
+            public override bool CanCompare(object x, object y)
+            {
+                return TypeHelper.CanCast<T>(x) && TypeHelper.CanCast<T>(y);
+            }
```

Comment:
What is the reason for introducing this override? (As far as I can tell it is overriding the same implementation in the parent class `GenericEqualityAdapter<T>`)

Reply:
> What is the reason for introducing this override? (As far as I can tell it is overriding the same implementation in the parent class `GenericEqualityAdapter<T>`)

Initially I was overriding all `CanCompare` operations as per idea in the original ticket. These were indeed useless.
I have added a commit to remove the new overloads again. Thanks for detecting this.